app/search/dataone.py

```diff
@@ -103,6 +105,7 @@ def combined_search(self, **kwargs):
         return self.execute_query(query, page_number)
 
     def convert_result(self, result):
+        global original_dict 
```

Comment:
I'm not sure this needs to be global - you could declare it here and set it to none instead, so that it's always defined, and then take out the `else` on line 157.